db/version_set.cc

```diff
@@ -3089,11 +3091,25 @@ unsigned VersionSet::PickCompactionLevel(bool* locked, bool seek_driven, bool* f
 		   * If the total amount of data in the level is less than threshold or if the next level has huge amount of
 		   * data compared to this level, compact this level.
 		   * */
-		  if (inter_level_ratio <= 25.0 || next_level_size_in_mb < FORCE_COMPACT_SIZE_THRESHOLD_IN_MB) {
-			  *force_compact = true;
-			  level = i;
-			  break;
-		  }
+          if (inter_level_ratio <= 25.0 || next_level_size_in_mb < FORCE_COMPACT_SIZE_THRESHOLD_IN_MB) {
+              
+              //Check if horizontal compaction is possible at all
```

Comment:
Could you add some more comments about what you are checking here?

A multi-line comment explaining what the previous bug would be useful.